addons/macrochanger/macrochanger.lua

```diff
@@ -26,143 +26,85 @@
+      settings.globaldisable = 1
+      windower.add_to_chat(17, 'All automated macro switching disabled.')
+    elseif args[2]:lower() == 'off' then
+      settings.globaldisable = 0
+      windower.add_to_chat(17, 'Automated macro switching enabled.')
+    end
+   elseif args[1]:lower() == 'help' then
+     windower.add_to_chat(17, 'MacroChanger Commands:')
+     windower.add_to_chat(17, 'disableall [on|off]')
+     windower.add_to_chat(17, '   on - Disables all automated macro switching')
+     windower.add_to_chat(17, '   off - Enables all automated macro switching not disabled individually')
+     windower.add_to_chat(17, '   Resets to what is stored in settings upon unloading of addon.  To Permanently change, please change the option in the settings file.')
+   end
 end
 
-windower.register_event('job change',function ()
+function job_change ()
 -- Could use the job ID passed into this function, but the addon would have to include the resources library
-	local job = windower.ffxi.get_player().main_job
-	local book = ''
-	local page = ''
-	if globaldisable == 0 then
-        if job and macros[job] then
-			book = macros[job].Book
-			page = macros[job].Page
-		end
+  local mjob = windower.ffxi.get_player().main_job
+  local sjob = windower.ffxi.get_player().sub_job
+  local job  = ''
+  local book = ''
+  local page = ''
+  if settings.globaldisable == 0 then
+    if mjob and sjob and settings.macros[(mjob..sjob):lower()] then
+      job = mjob..sjob
+    elseif mjob and settings.macros[(mjob):lower()] then
+      job = mjob
+    else
+      if sjob then job = mjob..' or '..mjob..sjob else job = mjob end
+      windower.add_to_chat(17, '     No Auto Macro Settings Available for '..job..'.')
+      return
+    end
+    page = settings.macros[(job):lower()].page
+    book = settings.macros[(job):lower()].book
+    if ((book == 'disabled') or (page == 'disabled')) then
+      windower.add_to_chat(17, '     Auto Macro Switching Disabled for ' .. job ..'.')
+    else
+      windower.add_to_chat(17, '     Changing macros to Book: ' .. book .. ' and Page: ' .. page .. '.  Job Changed to ' .. job)
+      windower.send_command('input /macro book '..book..';wait 0.2;input /macro set '..page..';')
+    end
+  else
+    windower.add_to_chat(17, '     Auto Macro Switching Disabled for All Jobs.')
+  end
+end
 
-		if ((book == 'disabled') or (page == 'disabled')) then
-			windower.add_to_chat(17, '                             Auto Macro Switching Disabled for ' .. job ..'.')
-		else
-			windower.add_to_chat(17, '                             Changing macros to Book: ' .. book .. ' and Page: ' .. page .. '.  Job Changed to ' .. job)
-			windower.send_command('input /macro book '..book..';wait 0.2;input /macro set '..page..';')
-		end
-	elseif globaldisable == 1 then
+function login (name)
+  local defaults = {
+    character = '',
+    globaldisable = 0,
+    macros = {
+      war = {book = 1, page = 1},
+    },
+  }
+  settings = config.load(defaults)
```

Comment:
Setting the default settings should happen on file scope, not on the login event. It will automatically reparse the settings when a new character logs in, so you don't need to do it manually.